app/core/file_processor/langchain_loader.py

```diff
@@ -46,11 +46,11 @@ def process_file_text(self,
                  label:str,
                  name: str = None,
                  metadata: dict = None) -> List[schema.Document]:
-        '''Uses langchain's CharacterTextSplitter to convert text contents into document format'''
+        '''Uses langchain's TokenTextSplitter to convert text contents into document format'''
         output_list = []
         loader = TextLoader(file)
         texts = loader.load()
-        text_splitter = CharacterTextSplitter(chunk_size=1000, chunk_overlap=0)
+        text_splitter = TokenTextSplitter(chunk_size=1000, chunk_overlap=200)
```

Comment:
I think in the long run we probably want to be more nuanced in how we do this (e.g., add the headings to the metadata). For now this looks good though!

Reply:
Also overlap of 50 or so might improve document distinctiveness so you don't get two documents that has very high similarity for overlapping chunks... although it is highly dependent on the query. Maximum marginal similarity vs. Cosine might work better here too to ensure more diverse documents but still high similarity.

Reply:
Yep. I'll change this to 50 for now.